### PR TITLE
fix: surface neutral OTP message on reset-password OTP screen

### PR DESCRIPTION
**Pull request by AmishaBisht**
## Summary

Frontend companion to the backend account-enumeration fix (**glific PR #5325**), plus the reset/login UX polish that came out of review.

### Why

The backend forgot-password (`send-otp`, `registration: false`) flow no longer discloses whether an account exists — it returns the **same** `200` for every phone number. Enumeration protection now comes from that identical response + identical navigation, **not** from any message text. This PR updates the frontend to match and cleans up the reset/login screens.

### Neutral, non-disclosing note
- The reset OTP screen shows a neutral note (**"If this number is registered, you'll receive an OTP on WhatsApp. Enter it below to reset your password."**) so a user whose number isn't registered understands why no OTP arrives — without the app ever confirming account existence.
- The note is **FE-owned static UI copy** (the backend still returns its own neutral message in the 200 body for API consumers). Rendered as a subtle, structured info note (icon + soft container), not a red error.
- The note **auto-hides after 15s** so it doesn't linger next to the resend warning.

### Phone number on the OTP screen
- The phone number entered on the previous screen is now **prepopulated** on the OTP screen (read from router state synchronously so Formik shows it), and remains **editable** — so a user can catch a typo and resend to the corrected number.

### Resend behaviour
- **Success:** briefly (2s) shows **"OTP sent successfully."** (green, with a check icon) so the user knows the OTP went out.
- **Failure:** shows **"Could not resend the OTP. Please try again in 30 seconds."** (pairs with the backend's per-IP rate limit in #5325); the info note is cleared so the two don't show at once.
- Resend now targets the **current (possibly edited)** phone value.

### Reset / login screen polish
- Removed the redundant **"New Password"** heading label (the field already has a "New Password" placeholder), matching the OTP field's style.
- Reset frame background is now **white**, like the login card.
- Fixed the odd **circular hover** on the OTP "resend" button.
- Auth **headings** (login + reset): kept on a **single line** (no lopsided wrap; larger font for the shorter headings), **centered**, colored the near-black brand green (`#073f24`, matching the "Glific" title), with **less whitespace above**.
- Added **spacing** between the New Password field and the Save button (was cramped).

### Cleanup
- Dropped dead imports (`setLogs`, an obsolete `eslint-disable`); guarded the resend callback with optional chaining.

## Testing

- `tsc --noEmit` clean; **all reset/login unit tests pass** (`src/containers/Auth/`), including new tests for the neutral note, prepopulated-phone resend, the resend success confirmation, the resend retry hint, and a fallback-branch case.
- Verified the full flow live in the browser (neutral note + auto-hide, prepopulated/editable phone, resend success/failure, white card, single-line centered headings, button spacing).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added an optional success/warning status note to auth flows.
  * Enhanced password reset with clearer OTP resend confirmations and retry guidance, including auto-hiding messages.
* **Bug Fixes**
  * Improved OTP resend reliability by preserving the phone number earlier and using it when resending.
  * Updated fallback messaging when backend resend info is missing or when no phone number is available.
* **Style**
  * Refined authentication layout/heading spacing and introduced styled info notes.
* **Tests**
  * Expanded coverage for OTP resend timing, messaging, and edge cases.

### CODE REVIEW

File: src/containers/Auth/ResetPassword/ResetPasswordPhone.tsx [hunk 1]
```diff
@@ -11,18 +11,25 @@ export const ResetPasswordPhone = () => {
...
+      // deliberately returns the same success response whether or not an account exists
+      // (to avoid account enumeration), so this is the only place the user learns that an
+      // OTP will arrive *only if* the account exists.
+      otpInfoMessage,
     };
     return <Navigate to="/resetpassword-confirmotp" state={stateObject} />;
   }
 
   const onSubmitPhone = (data: any) => {
     sendOTP(data.phoneNumber, undefined, 7000)
```
Comment (rvignesh89) [inline src/containers/Auth/ResetPassword/ResetPasswordPhone.tsx:26]: handle that case when resendOTP return an error to try again in 30 seconds

File: src/containers/Auth/ResetPassword/ResetPasswordConfirmOTP.tsx [hunk 2]
```diff
@@ -19,11 +19,13 @@ export const ResetPasswordConfirmOTP = () => {
   const { t } = useTranslation();
   const location = useLocation();
   const [phoneNumber, setPhoneNumber] = useState('');
```
Comment (rvignesh89) [inline src/containers/Auth/ResetPassword/ResetPasswordConfirmOTP.tsx]: lets prepopulate with the same phone number entered in the screen before. But allow them to edit and resend OTP

File: src/containers/Auth/ResetPassword/ResetPasswordConfirmOTP.tsx [hunk 3]
```diff
@@ -19,11 +19,13 @@ export const ResetPasswordConfirmOTP = () => {
   const { t } = useTranslation();
```
Comment (rvignesh89) [inline src/containers/Auth/ResetPassword/ResetPasswordConfirmOTP.tsx:24]: Image

the background of reset password frame should be white like login

File: src/containers/Auth/ResetPassword/ResetPasswordConfirmOTP.tsx
Comment (rvignesh89) [inline src/containers/Auth/ResetPassword/ResetPasswordConfirmOTP.tsx:108]: Image

resend onhover looks weird

**Review by rvignesh89 — CHANGES_REQUESTED** [review]

File: src/containers/Auth/ResetPassword/ResetPasswordConfirmOTP.tsx [hunk 4]
```diff
@@ -8,24 +8,38 @@ import { RESET_PASSWORD } from 'config';
...
-import setLogs from 'config/logs';
 import { Auth } from '../Auth';
 
+// Neutral, non-disclosing note: shown to everyone regardless of whether the account exists,
+// so a user whose number isn't registered understands why no OTP arrives (and can fix a typo
+// in the editable phone field above) without the API ever confirming account existence.
+const OTP_INFO_NOTE =
+  "If this number is registered, you'll receive an OTP on WhatsApp. Enter it below to reset your password.";
+const OTP_INFO_NOTE_TIMEOUT = 15000;
+const RESEND_SUCCESS_TIMEOUT = 2000;
```
Comment (rvignesh89) [inline src/containers/Auth/ResetPassword/ResetPasswordConfirmOTP.tsx]:
```suggestion
const RESEND_SUCCESS_TIMEOUT = 5000;
```

**Review by rvignesh89 — APPROVED** [review]

File: src/containers/Auth/ResetPassword/ResetPasswordConfirmOTP.tsx [hunk 5]
```diff
@@ -8,24 +8,39 @@ import { RESET_PASSWORD } from 'config';
...
 import { Auth } from '../Auth';
 
+// Neutral, non-disclosing note: shown to everyone regardless of whether the account exists,
+// so a user whose number isn't registered understands why no OTP arrives (and can fix a typo
+// in the editable phone field above) without the API ever confirming account existence.
+const OTP_INFO_NOTE =
+  "If this number is registered, you'll receive an OTP on WhatsApp. Enter it below to reset your password.";
+const OTP_INFO_NOTE_TIMEOUT = 15000;
+const RESEND_SUCCESS_TIMEOUT = 5000;
+const RESEND_RATE_LIMIT_MESSAGE = 'An OTP was just sent. Please try again in 30 seconds.';
```
[the remark this answers is not in the document]
Comment (AmishaBisht) [reply src/containers/Auth/ResetPassword/ResetPasswordConfirmOTP.tsx:21]: Fixed in 91f744478 — OTP_INFO_NOTE, 'OTP sent successfully.', and the resend rate-limit message now go through t() (moved useTranslation() above the state so the initial note is translated too).

File: src/containers/Auth/ResetPassword/ResetPasswordConfirmOTP.tsx [hunk 6]
```diff
@@ -36,8 +51,26 @@ export const ResetPasswordConfirmOTP = () => {
...
+        setTimeout(() => {
+          setInfoVariant(undefined);
+          setInfoMessage('');
+        }, RESEND_SUCCESS_TIMEOUT);
+      })
+      .catch((error) => {
+        const backendMessage = error?.response?.data?.error?.message;
+        setInfoVariant('warning');
+        setInfoMessage(backendMessage || RESEND_RATE_LIMIT_MESSAGE);
+      });
```
[the remark this answers is not in the document]
Comment (AmishaBisht) [reply src/containers/Auth/ResetPassword/ResetPasswordConfirmOTP.tsx:91]: Fixed in 91f744478 — handleResend now clears any pending successTimerRef at the start (and nulls it) before scheduling a new one, so an older success timer can't clear a newer success/warning message. Unmount cleanup is unchanged.
